robot/Cumulus/tests/browser/levels/create_edit_delete_level.robot

```diff
@@ -4,119 +4,115 @@ Resource        robot/Cumulus/resources/NPSP.robot
+    Enter Level Dd Values
+    ...                                                 Target=Contact
+    ...                                                 Source Field=Smallest Gift
+    ...                                                 Level Field=Level
+    ...                                                 Previous Level Field=Previous Level
+    Click Button                                        Save
+    Current Page Should be                              Details    Level__c
+    ${level_id} =                                       Save Current Record ID For Deletion  Level__c
+    Set Global Variable                                 ${level_id}
+    Go To Page                                          Details
+        ...                                             Level__c
+        ...                                             object_id=${level_id}
+
+    Wait Until Loading Is Complete
+    Navigate To And Validate Field Value                Minimum Amount (>\=)    contains    ${min_amount}
+    Navigate To And Validate Field Value                Maximum Amount (<)      contains    ${max_amount}
+
+    Click Show More Actions Button                      Edit
+    Current Page Should Be                              Custom                             Level__c
+
+
     Enter Level Values
-    ...            Minimum Amount=0.01
-    ...            Maximum Amount=0.99
-    Set Focus To Element   xpath: //input[@value='Save']
-    Click Button  Save
-    #adding a workaround to go back to levels tab due to core issue
-    Current Page Should Be     Details    Level__c
-    Go To Page         Details         Level__c        object_id=${level_id}
-    Navigate To And Validate Field Value   Minimum Amount (>\=)    contains    0.01
-    Navigate To And Validate Field Value   Maximum Amount (<)     contains    0.99
-    Navigate To And Validate Field Value    Source Field    contains    npo02__SmallestAmount__c
-
-3 Validate Level Assignment in Batch Job
-    [tags]  unstable
-    # --------------------------------
-    # Modify the SmallestGift field to allow the level to be applied
-    # --------------------------------
-    &{contact} =  API Create Contact
-    Set Global Variable     ${contact_id}       &{contact}[Id]
-    Go To Page    Details    Contact       object_id=${contact_id}
-    Scroll Element Into View    text:Donation Totals
-    Click Button       title:Edit Smallest Gift
-    Wait For Locator  record.footer
-    Populate Field          Smallest Gift     0.75
-    Click Record Button     Save
+    ...                                                 Minimum Amount=${minamount_to_edit}
+    ...                                                 Maximum Amount=${maxamount_to_edit}
+    Enter Level Dd Values
+    ...                                                 Source Field=Smallest Gift
+
+    Click Button                                        Save
```

Comment:
after hitting save, it would be better if u can add some kind of validation to make sure the next page is loaded or save button is not visible. I have noticed n some cases it takes longer to save (especially when the system is slow)and actually save won't happen if u navigate immediately away from the page.

Reply:
Adding logic after Save